### ISSUE TITLE
[Flake] macOS build takes over 1 hour to upload (useless) artifacts

### ISSUE THREAD

**Issue by coryan** [googleapis/google-cloud-cpp#4507]
This build usually takes a few minutes, in this case the build was stuck for over one hour in these steps:

```
[ID: 8993353] Build finished after 178 secs, exit value: 0


Warning: Permanently added 'localhost' (ECDSA) to the list of known hosts.
```

Note that this is after our build script has finished, I killed the build, which shows Kokoro trying to fetch over 800MiB of artifacts:

```
rsync: connection unexpectedly closed (796110846 bytes received so far) [receiver]
[12:16:25] Collecting build artifacts from build VM
Received signal: 15, build might get interrupted.
Received signal: 15, build might get interrupted.
rsync error: received SIGINT, SIGTERM, or SIGHUP (code 20) at rsync.c(642) [generator=3.1.3]
rsync error: error in rsync protocol data stream (code 12) at io.c(226) [receiver=3.1.3]
Warning: Permanently added 'localhost' (ECDSA) to the list of known hosts.
[13:16:49] Kokoro builder finished
```

https://source.cloud.google.com/results/invocations/896f01fc-aa98-41c0-a4fd-0ca3008d6c34/targets

I think we need to erase the build artifacts before exiting our builds, like we do for Linux and Windows.

**Comment by coryan** [comment]
Not planning on other changes after #4508. We should close this bug by 2020-10-13 unless the flake repeats before then.

**Comment by coryan** [comment]
No repeats in 90d, the fixes in #4508 seem to have minimized the problem to the point of irrelevancy.
